### PR TITLE
Feature

### PR DESCRIPTION
**Pull request by kmakeev**
small changes and results to LeaderBoard

### CODE REVIEW

File: Nn/tf2nn.py [hunk 1]
```diff
@@ -64,7 +64,7 @@ class actor_mu(Model):
     def __init__(self, vector_dim, output_shape, name, hidden_units, *, visual_net):
         super().__init__(visual_net, name=name)
         self.net = mlp(hidden_units, output_shape=output_shape, out_activation='tanh')
-        self.init_or_run(tf.keras.Input(shape=vector_dim + self.visual_net.hdim), tf.keras.Input)
+        self.init_or_run(tf.keras.Input(shape=vector_dim + self.visual_net.hdim))
```
Comment (StepNeverStop) [inline Nn/tf2nn.py]: keep this

File: common/agent.py [hunk 2]
```diff
@@ -1,4 +1,4 @@
-import os
+import os, sys
```
Comment (StepNeverStop) [inline common/agent.py]: refuse

File: common/agent.py [hunk 3]
```diff
@@ -19,9 +19,10 @@ def ShowConfig(config):
 
 def UpdateConfig(config, file_path, key_name='algo'):
     _config = load_config(file_path)
+    values = _config[key_name]
```
Comment (StepNeverStop) [inline common/agent.py]: refuse

File: common/agent.py [hunk 4]
```diff
@@ -19,9 +19,10 @@ def ShowConfig(config):
...
     try:
-        for key in _config[key_name]:
-            config[key] = _config[key]
+        for key in values:
+            config[key] = values[key]
```
Comment (StepNeverStop) [inline common/agent.py]: refuse

File: common/agent.py [hunk 5]
```diff
@@ -325,7 +326,7 @@ def gym_train(self):
                     s=state[0],
                     visual_s=state[1],
                     a=action,
-                    r=reward,
+                    r=reward.astype('float32'),
```
Comment (StepNeverStop) [inline common/agent.py]: refuse

File: common/agent.py [hunk 6]
```diff
@@ -412,7 +413,7 @@ def gym_random_sample(self, steps):
                 s=state[0],
                 visual_s=state[1],
                 a=action,
-                r=reward,
+                r=reward.astype('float32'),
```
Comment (StepNeverStop) [inline common/agent.py]: refuse

File: common/agent.py [hunk 7]
```diff
@@ -424,7 +425,7 @@ def gym_random_sample(self, steps):
 
     def gym_evaluate(self):
         max_step = int(self.train_args['max_step'])
-        max_eval_episode = int(self.train_args['max_eval_eposide'])
+        max_eval_episode = int(self.train_args['max_eval_episode'])
```
Comment (StepNeverStop) [inline common/agent.py]: keep this

File: common/agent.py [hunk 8]
```diff
@@ -475,7 +476,7 @@ def gym_no_op(self):
                 s=state[0],
                 visual_s=state[1],
                 a=action,
-                r=reward,
+                r=reward.astype('float32'),
```
Comment (StepNeverStop) [inline common/agent.py]: refuse

File: gym_wrapper/wrapper_linux.py [hunk 9]
```diff
@@ -89,7 +89,8 @@ def get_env(kwargs):
         self._get_render_index(render_mode)
 
     def _initialize(self, env):
-        assert isinstance(env.observation_space, (Box, Discrete)) and isinstance(env.action_space, (Box, Discrete)), 'action_space and observation_space must be one of available_type'
+        assert isinstance(env.observation_space, (Box, Discrete)) \
+               and isinstance(env.action_space, (Box, Discrete, Tuple)), 'action_space and observation_space must be one of available_type'
```
Comment (StepNeverStop) [inline gym_wrapper/wrapper_linux.py]: keep this

File: gym_wrapper/wrapper_win.py [hunk 10]
```diff
@@ -83,7 +83,8 @@ def get_env(config: Config):
         self._get_render_index(render_mode)
 
     def _initialize(self, env):
-        assert isinstance(env.observation_space, (Box, Discrete)) and isinstance(env.action_space, (Box, Discrete)), 'action_space and observation_space must be one of available_type'
+        assert isinstance(env.observation_space, (Box, Discrete)) \
```
Comment (StepNeverStop) [inline gym_wrapper/wrapper_win.py]: keep this

**Comment by StepNeverStop** [comment]
Regisiter must be finished before submit a record of leader board. At https://github.com/openai/gym/wiki/Leaderboard

File: common/agent.py (hunk 2 above)
Comment (kmakeev) [reply common/agent.py]: sys use in
```
27        sys.exit() 
```

File: common/agent.py (hunk 5 above)
Comment (kmakeev) [reply common/agent.py]: If the environment returns a reward in int,
there is an error in the calculations at least here:

replay_buffer.py
```
206 q [j] [3] + = data [3] * (self.gamma ** (_len - j))
```

File: common/agent.py (hunk 4 above)
Comment (kmakeev) [reply common/agent.py]: The current code leads to an error, since the configuration dictionary now contains section keys, rather than parameter names:
``` algo:
   actor_lr: 0.0005
   batch_size: 1024
   beta: 0.001
   critic_lr: 0.001
   epoch: 4
   gamma: 0.99
```

**Comment by kmakeev** [comment]
> Regisiter must be finished before submit a record of leader board. At https://github.com/openai/gym/wiki/Leaderboard

I can not give a link to the result until my PR is accepted by him.

File: common/agent.py (hunk 5 above)
Comment (StepNeverStop) [reply common/agent.py]: that's a bug, but I'd rather fix it in`gym_wrapper/wrapper_win.py` line 201. Let environment wrapper control anything about environment. I do not want any data type convertion occured in the section of training, because it's going to be messy to maintain this project.
